### ISSUE TITLE
Improving mask contour

### ISSUE THREAD

**Issue by acaelles97** [zjhuang22/maskscoring_rcnn#86]
First I would like to thank you for your awesome work. I am currently using mask scoring and obtaining really good results, but I have a problem related with the contour of some  masks. If you compute the IoU of the predicted mask with the GT mask, the value is high, as it is almost overlapping the target object but if you take a closer look, the points from the contour are not really correctly delimiting the object . To illustrate this, if an object has a completely straight shape, the contour from the mask sometimes has like a wave shape instead of a perfect straight line.
This effect is not present  on the object masks from my training set, so I guess it might be a problem related with the architecture of the framework or maybe i am not adjusting some hyper-parameter in a correct way. I am also training in High Resolution images (2560x2048) so it can not be a problem from the image resolution itself.    
I have been thinking on a possible solution, and maybe i could add an extra loss to the framework to compute how good the contour of the predicted mask is compared to the GT mask. This should avoid not penalizing masks when the IoU is really high but the contour is not adjusting so well.
What do you think that may be causing this problem? Can you give me any piece of advise on possible solutions?

Thanks a lot!!
